external-crates/move/crates/move-compiler/src/parser/ast.rs

```diff
@@ -1281,6 +1359,70 @@ impl AstDebug for FriendDecl {
     }
 }
 
+impl AstDebug for EnumDefinition {
+    fn ast_debug(&self, w: &mut AstWriter) {
+        let EnumDefinition {
+            attributes,
+            loc: _loc,
+            abilities,
+            name,
+            type_parameters,
+            variants,
+        } = self;
+        attributes.ast_debug(w);
+
+        if !abilities.is_empty() {
+            w.write("[");
+            w.list(abilities, " ", |w, ab_mod| {
+                ab_mod.ast_debug(w);
+                false
+            });
+            w.write("]");
+        }
+
+        w.write(&format!(" enum {}", name));
+        type_parameters.ast_debug(w);
+        w.block(|w| {
+            if let Some(v) = variants.first() {
+                v.ast_debug(w)
+            };
+            w.write(",");
+            if variants.len() > 1 {
+                w.comma(variants.iter().skip(1), |w, v| {
```

Comment:
These should all be with `w.list` which will manage the new line for you (via the bool)